model_benchmarking/preprocessing_scripts/label_quailty_check.py

```diff
@@ -0,0 +1,438 @@
+import os
+import torch
+import json
+import random
+import shutil
+import utilities
+import numpy as np
+import pandas as pd
+import torch.nn as nn
+import scipy.optimize
+from pycocotools.coco import COCO
+from sklearn.metrics import multilabel_confusion_matrix
+from sklearn.metrics import cohen_kappa_score
+from sklearn.metrics import f1_score
+from sklearn.metrics import matthews_corrcoef
+
+def sample_data(data_dir, if_training, target_dir, json_file_name=None):
+    if if_training:
+        json_file_name = "animal_count_key.json"
+        for directory in os.scandir(data_dir):
+            if directory.is_dir():
+                directory_path = directory.path + "/"
+                print("\nSample data from directory: ", directory_path)
+                temp_data, temp_lab = sample_data_from_folder(directory_path, json_file_name, target_dir, True)
+                print("Number of images sampled: ", len(temp_data), ' Number of labels sampled: ', len(temp_lab))
+    else:
+        sample_data_from_folder(data_dir, json_file_name, target_dir, False)
+
+def sample_data_from_folder(data_dir, json_file_name, target_dir, if_training):
+    if if_training:
+        coco = COCO(data_dir + json_file_name)
+        images = coco.loadImgs(coco.getImgIds())
+        sorted_images = utilities.get_sorted_images(images)
+        
+        data_ids, data_paths, file_names, labels = [], [], [], []
+        
+        for image in sorted_images:
+            file_name = image["file_name"]
+            file_path = data_dir + file_name
+            
+            annotation_id_list = coco.getAnnIds(imgIds=[image["id"]])
+            annotation_list = coco.loadAnns(annotation_id_list)
+            
+            if os.path.isfile(file_path):
+                data_ids.append(image["id"])
+                data_paths.append(file_path)
+                file_names.append(file_name)
+                labels.append(annotation_list)
+            else:
+                print("No file found for: ", file_path)
+                continue
+        
+        print("Number of images found:", len(data_paths))
+        
+        if "caltech" in data_dir:
+            sampled_indices = sorted(random.sample(range(0, len(data_paths)), 300))
```

Comment:
The hardcoded value of 300 is used because the caltech folder has over 75,000 images. Sampling 5% will result in more than 3,750 images for label quality checks, which exceeds our expected workload. However, since only ~6,000 images from caltech were actually used for model training, I manually set the sample size to 300, which is 5% of the 6,000 images.

Reply:
Shouldn't we just be checking the 6000 images that were actually used for model training then? And resampling 5% of those 6000 images. The utilities.fetch_data() function should allow for this.